### ISSUE TITLE
[QUESTION] In the rails README example, does it make sense to cache it?

### ISSUE THREAD

**Issue by aesyondu** [ztoben/assets-webpack-plugin#212]
I followed the [rails low-level caching guide](https://guides.rubyonrails.org/caching_with_rails.html#low-level-caching) and came up with the following:
```ruby
def get_webpack_assets
  Rails.cache.fetch("webpack-assets.json", expires_in: 12.hours) do
    path = Rails.root.join('app', 'views', 'webpack-assets.json')
    file = File.read(path)
    JSON.parse(file)
  end
end

def script_for(bundle)
  json = get_webpack_assets
  json[bundle]['js']
end

def style_for(bundle)
  json = get_webpack_assets
  json[bundle]['css']
end
```

I'm not that familiar with rails caching, is there any benefit to doing it this way?

EDIT: I made it more robust
```ruby
  def get_webpack_assets
    webpack_assets_path = Rails.root.join('app', 'views', 'webpack-assets.json')
    webpack_assets_key = Digest::MD5.file(webpack_assets_path).hexdigest
    Rails.cache.fetch(webpack_assets_key, expires_in: 12.hours) do
      file = File.read(webpack_assets_path)
      JSON.parse(file)
    end
  end

  def script_for(bundle)
    json = get_webpack_assets
    if json.key?(bundle) && json[bundle].key?("js")
      json[bundle]["js"]
    else
      logger.error "JavaScript bundle '#{bundle}' does not exist"
      ""
    end
  end

  def style_for(bundle)
    json = get_webpack_assets
    if json.key?(bundle) && json[bundle].key?("css")
      json[bundle]["css"]
    else
      logger.error "CSS bundle '#{bundle}' does not exist"
      ""
    end
  end
```

**Comment by ztoben** [comment]
Can't say I'm much of a rails guy, so if someone else has some input I'll leave this open for a while for comments. I wouldn't imagine it makes much sense to cache it, but that's just my feeling.

**Comment by aesyondu** [comment]
**TLDR; No it doesn't make sense.**

I just did a basic performance test:
```ruby
# WITH CACHING
[2] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.0005912252000071022
[3] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.0005827203999804624
[4] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.000584537399985129

# WITHOUT CACHING
[2] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.000311480500027028
[3] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.00029823210001086407
[4] pry(main)> (1..10000).map { |i| Benchmark.measure { get_webpack_assets }.real }.sum/10000
=> 0.0002968197000096552
```

I found the caching approach taking longer to be unintuitive. I guess it makes sense considering my webpack-manifest.json file is 7.5 kilobytes.
